src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Text,
+  Image,
+  Container,
+  Divider,
+  Title,
+  Badge,
+  Group,
+  Button,
+  Space,
+} from "@mantine/core";
+import { useState, useEffect } from "react";
+import { getSingleArticle, updateVote } from "../../api.js";
+import { useParams } from "react-router-dom";
+import Loading from "../Loading";
+import Error from "../Error";
+import Comments from "./Comments.jsx";
+
+const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [error, setError] = useState(undefined);
```

Comment:
Undefined is an unintentional absence of something. If we intend to have an empty variable, we would use null.

In this case error state could change from null to err   -   or   -   flip from false to true